### ISSUE TITLE
Support OAuth2 access_token authorization

### ISSUE THREAD

**Issue by rehandalal** [mozilla/normandy#1245]
We need to find a way to use Auth0 access tokens to access the API without the current redirect initiated by Nginx.

**Comment by peterbe** [comment]
I think and hope we can use [mozilla-django-oidc](https://mozilla-django-oidc.readthedocs.io/en/stable/) but instead of doing the redirect dance by `code` we'll send Auth0 the `access_token` to verify it. This will sit as a middleware protecting the Django REST framework so that `request.user` exists. 

This might be a good time to also do https://github.com/mozilla/normandy/issues/1244 which is to set `request.user.first_name` and `request.user.last_name` from the Auth0 profile.

**Comment by mostlygeek** [comment]
@peterbe look into if we can get an JWT access_token instead of the opaque string.  

We will need to support multiple clients (Console, Experimenter, etc) talking to the API.  From [Auth0's docs](https://auth0.com/docs/api-auth/why-use-access-tokens-to-secure-apis) this is what a JWT access_token should look like: 

```js
{
  "iss": "https://YOUR_AUTH0_DOMAIN/",
  "sub": "auth0|123456",
  "aud": [
    "my-api-identifier",
    "https://YOUR_AUTH0_DOMAIN/userinfo"
  ],
  "azp": "YOUR_CLIENT_ID",
  "exp": 1489179954,
  "iat": 1489143954,
  "scope": "openid profile email address phone read:appointments email"
}
```

We don't want to trust any issuer. There should be a whitelist of issuers (`iss:`) that we accept tokens from.  

The endpoints in `aud:` should give us the profile information we need (email, name, etc).

**Comment by peterbe** [comment]
Here's the code that proved the prototype: https://github.com/mozilla/normandy/compare/mozilla:master...peterbe:auth-by-bearer-authorization-header-fixes-1245

* [ ] Tune the code and resolve all `XXX ...` comments
* [ ] Unit tests
* [ ] Wonder if this should only work for certain paths like `/api/v2`. Some whitelist might be a good idea. 

Also, we have two other concurrent issues to solve. The delivery console needs to correctly send the access_token. 
And our Nginx Lua code doesn't allow any requests through (even if the header already has a `Authorization`). We need to find a way to bypass that. Ideally with some conditionals rather than having to edit the Lua code.

**Comment by peterbe** [comment]
Heck, I made a [WIP PR](https://github.com/mozilla/normandy/pull/1322) instead. We can chat there.

**Comment by mythmon** [comment]
Fixed by #1322

**Comment by peterbe** [comment]
Sigh bors. I'm sure I figured this out. You have to says "fixes XXXX" not "fixes #XXXX" for the issue to get closed. 
I did that in https://github.com/mozilla/normandy/pull/1322/commits but still didn't close the issue.
